cozy/solver.py

```diff
@@ -1,3 +1,7 @@
+## TODO: This comment says what the file does.  It happens to use a solver
+## internally, but I would rename the file from `solver.py` to make the
+## name relate to the specified functionality rather than the
+## implementation.
```

Comment:
I would call any code that checks satisfiability or validity of expressions a "solver". This module really is a solver for Cozy expressions.

You are right that we should remove instances of "SMT" from the docstring here, since that does feel like an internal detail.